src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -153,6 +154,62 @@ public async Task<HttpResponseMessage> HandleRequestAsync(HttpRequestMessage req
                     return await this.HandleStartOrchestratorRequestAsync(request, functionName, instanceId);
                 }
 
+                i = path.IndexOf(EntitiesControllerSegment, StringComparison.OrdinalIgnoreCase);

```

Comment:
In general, this method is starting to get very difficult to read. @cgillum, at some point we should consider refactoring this method.

When I had to do a similar set of APIs in a functions extension, I wrote it as a series of if statements that were effectively a series of if/else-if statements where each condition checked if the requestUrl matched the expected pattern and that the request method matched expectations. This is almost certainly not as effecient, but it was fairly readable in my opinion.

As we keep adding more and more methods, I think we need some easily expandable and readable approach to handle this problem.

Reply:
I agree. At a minimum, create subhandlers for each prefix to avoid a super-long function body.

Reply:
I agree we should refactor this at some point. I think I went down this path originally because I figured it would be simple and highly optimized, but it ended up being not very maintainable. I opened an issue for tracking: https://github.com/Azure/azure-functions-durable-extension/issues/735